modules/ca/src/client/udpiiu.cpp

```diff
@@ -725,6 +725,33 @@ bool udpiiu :: searchRespAction (
         serverAddr.ia.sin_addr = addr.ia.sin_addr;
     }
 
+#ifdef MIN_IOC_CA_VER_WHEN_REDIRECT
+    // Try to detect the nameserver redirecting us to a different IP
+    if (memcmp(&addr.ia.sin_addr, &serverAddr.ia.sin_addr, sizeof(struct in_addr)) != 0) {
+
```

Comment:
Does it make sense to avoid memcmp() here and use something similar
as coded in repeater.cpp ?

`inline bool repeaterClient::identicalAddress ( const osiSockAddr &fromIn )
{
    if ( fromIn.sa.sa_family == this->from.sa.sa_family ) {
        if ( fromIn.ia.sin_port == this->from.ia.sin_port) {
            if ( fromIn.ia.sin_addr.s_addr == this->from.ia.sin_addr.s_addr ) {
                return true;
            }
        }
    }
    return false;
}

`

Reply:
I was not aware that repeater.cpp already does this. I used memcmp because of concern for sockaaddr possibly having platform dependent implementation (POSIX, BSD, WIN32). If that has been addressed (and used in repeater), I'm happy to change it to the expanded version as it's more intuitive albeit longer.

Reply:
A couple of points.

IPv4 addresses `sockaddr_in::in_addr::s_addr`, aka `in_addr_t` can be compared as an integer.  No need for `memcmp()`.

Also, while not immediately relevant.  Be careful about comparing the whole `sockaddr_in` with `memcmp()` as the `sockaddr_in::sin_zero` member should not be compared (it may not always be zero...).  I found this out the hard way with the RTEMS legacy network stack...